tests/unit/test_config.py

```diff
@@ -100,6 +100,12 @@
 """
 
 
+@pytest.fixture()
+def default_files(monkeypatch):
+    """Overrides mocked default files from conftest.py as we have our own mocks here."""
+    monkeypatch.setattr(gitlab.config, "_DEFAULT_FILES", config._DEFAULT_FILES)
+
```

Comment:
Why is this function in this file? I'm confused how it is used.

Reply:
The global conftest now removes all `_DEFAULT_FILES` so the default behavior in tests is that it won't find any of these files.

In `test_config.py` we have tests that expect `_DEFAULT_FILES` to match our production list, and have their own mocks that mock opening these files instead. So by reverting to this list in the module we let the existing test mocks take care of it. The override works kind of like python's LEGB for variables, the module level fixture overrides the root one.

Maybe a better way long-term would be to completely rework the tests in this module to mock things differently, but this resulted in the smallest diff.

Reply:
Okay. Seems to be because it is being overridden: https://docs.pytest.org/en/latest/how-to/fixtures.html#overriding-fixtures-on-various-levels

Would be nice if `autouse=True` is in the function definition, for understand-ability. I realize it isn't needed.

WDYT?

Reply:
Oh yes, that was actually supposed to be there :facepalm: added now. And yes, that links explains the override much better than my mini-novel :grin: